### PR TITLE
[Relax][TFLite] Add test coverage for Reduction operations (#18971)

### PR DESCRIPTION
**Pull request by rknastenka**
Closes part of #18971

---

## Description
This PR improves the test coverage of the TFLite frontend to Relax converter by adding comprehensive tests for the **"Reductions" group.**

**Operations covered:**
* `SUM` (`tf.reduce_sum`)
* `MEAN` (`tf.reduce_mean`)
* `REDUCE_MAX` (`tf.reduce_max`)
* `REDUCE_MIN` (`tf.reduce_min`)
* `REDUCE_PROD` (`tf.reduce_prod`)

## Changes made:
* Added a parameterized testing function (`test_reduction_ops`) to cover combinations of the aforementioned reduction operators.
* Covered a variety of axis configurations, including positive scalars, lists of axes, negative indices, and `None` (global reduction).
* Tested with different combinations of the `keepdims` flag (`True`/`False`) and dtypes (`float32`/`int32`).
* Handled the representation of global reductions (`axis=None`) in the expected Relax IR Module `_make_reduce_expected` utility by expanding it to all input axes, perfectly mirroring the frontend's output graph structure (`list(range(len(input_shape)))`).

## Testing
```bash
pytest tests/python/relax/test_frontend_tflite.py::test_reduction_ops
```
Screenshot 2026-04-08 102859

### CODE REVIEW

**Review by tlopex — APPROVED** [review]
LGTM
